src/account/operations/transfer/high_level/sweep_outputs.rs

```diff
@@ -0,0 +1,242 @@
+            Output::Alias(alias_output) => alias_output.clone(),
+            _ => unreachable!("We already checked that it's an alias output"),
+        };
+
+        Ok((*output_id, alias_output))
+    }
+
+    pub(crate) async fn sweep_address_outputs(
+        &self,
+        address: Address,
+        remainder_address: AddressWrapper,
+    ) -> crate::Result<()> {
+        let query_parameters = vec![
+            QueryParameter::Address(address.to_bech32(remainder_address.bech32_hrp())),
+            QueryParameter::HasExpirationCondition(false),
+            QueryParameter::HasTimelockCondition(false),
+            QueryParameter::HasStorageDepositReturnCondition(false),
+        ];
+
+        let alias_output_ids = self.client.aliases_output_ids(query_parameters.clone()).await?;
+        let basic_output_ids = self.client.output_ids(query_parameters.clone()).await?;
+        let nft_output_ids = self.client.nfts_output_ids(query_parameters).await?;
+
+        let alias_outputs = self.client.get_outputs(alias_output_ids).await?;
+        let basic_outputs = self.client.get_outputs(basic_output_ids).await?;
+        let nft_outputs = self.client.get_outputs(nft_output_ids).await?;
+
+        let mut output_ids = Vec::new();
+        let mut outputs = Vec::new();
+
+        for mut output_response in alias_outputs
+            .into_iter()
+            .chain(basic_outputs.into_iter())
+            .chain(nft_outputs.into_iter())
+        {
+            if output_response.is_spent {
+                continue;
+            }
+
+            match &mut output_response.output {
+                OutputDto::Basic(output_dto) => {
+                    Self::replace_unlock_conditions(&mut output_dto.unlock_conditions, &remainder_address.inner);
+                }
+                OutputDto::Alias(output_dto) => {
+                    Self::replace_unlock_conditions(&mut output_dto.unlock_conditions, &remainder_address.inner);
+                }
+                OutputDto::Nft(output_dto) => {
+                    Self::replace_unlock_conditions(&mut output_dto.unlock_conditions, &remainder_address.inner);
+                }
+                OutputDto::Treasury(_) | OutputDto::Foundry(_) => continue,
+            }
+
+            let transaction_id = TransactionId::from_str(&output_response.transaction_id)?;
+            let output_id = OutputId::new(transaction_id, output_response.output_index)?;
+            let output = Output::try_from(&output_response.output)?;
+
+            output_ids.push(output_id);
+            outputs.push(output);
+
+            if output_ids.len() == (OUTPUT_COUNT_MAX - 1) as usize {
+                self.send_sweep_transaction(address, output_ids.drain(..), outputs.drain(..))
+                    .await?;
+            }
+        }
+
+        if !output_ids.is_empty() {
+            self.send_sweep_transaction(address, output_ids.drain(..), outputs.drain(..))
+                .await?;
+        }
+
+        Ok(())
```

Comment:
Can you return the transaction ids of the sweeping transactions here?

Reply:
Will do that